peers.py

```diff
@@ -34,6 +34,13 @@ def peers_changed(self):
         if conv.get_remote('unit_name'):
             conv.set_state('{relation_name}.joining')
 
+    @hook('{peers:etcd}-relation-broken')
+    def peers_going_away(self):
+        conv = self.conversation()
```

Comment:
Couldn't you scope the conversation to UNIT right here and do this in '-departed'?